soc/st/stm32/stm32mp13x/soc.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for STM32MP13 processor
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/linker/linker-defs.h>
+
+#include <stm32_ll_bus.h>
+#include <cmsis_core.h>
+
+/**
+ * @brief Perform basic hardware initialization at boot.
+ *
+ * This needs to be run from the very beginning.
+ * So the init priority has to be 0 (zero).
+ *
+ * @return 0
+ */
+#define VECTOR_ADDRESS ((uintptr_t)_vector_start)
+
+void relocate_vector_table(void)
+{
+	write_sctlr(read_sctlr() & ~HIVECS);
+	write_vbar(VECTOR_ADDRESS & VBAR_MASK);
+	barrier_isync_fence_full();
+}
+
+static int stm32mp13_a7_init(void)
+{
+	/* Update CMSIS SystemCoreClock variable (HCLK) */
+	SystemCoreClock = 1000000000U;
+
+	__ASM volatile(
+
+		/* Read CP15 System Control register */
+		"MRC     p15, 0, R0, c1, c0, 0\n"
+
+		/* Clear TE bit to take exceptions in Thumb mode to fix the DDR init */
+		"BIC     R0, R0, #(0x1 << 30)\n"
+
+		/* Write value back to CP15 System Control register */
+		"MCR     p15, 0, R0, c1, c0, 0\n"
+		"ISB\n"
+		"LDR     R0, =_vector_table\n"
+		"MCR     p15, 0, R0, c12, c0, 0\n");
```

Comment:
Avoid inline assembly if CMSIS or Zephyr equivalents are available:
```suggestion
	write_sctlr(read_sctlr() & ~SCTLR_TE_Msk);
	barrier_isync_fence_full();

	write_vbar(VECTOR_ADDRESS);
```
(Also, is the `write_vbar` really necessary here?)